tasks/contract.ts

```diff
@@ -95,3 +97,46 @@ async function generate({ x, y }: { x: number; y: number }, hre: HardhatRuntimeE
 
   return ret;
 }
+
+task("request:coords", "request a reveal")
+  .addParam("x", "x value", undefined, types.int)
+  .addParam("y", "y value", undefined, types.int)
+  .setAction(requestCoords);
+
+async function requestCoords({ x, y }: { x: number; y: number }, hre: HardhatRuntimeEnvironment) {
```

Comment:
It seems that HH `type.int` doesn't accept negatives, so can these just be strings? Would also need to be updated for the generate task.

Reply:
bastards. int means int not uint, Ill put them back to string (and theyll still be validated by the bignumberish in the contract call) because the only other numeric type they have is float